### PR TITLE
Do not use "nil" to refer to `()`

### PR DESCRIPTION
**Pull request by rossmacarthur**
"nil" is not used in the [book](https://doc.rust-lang.org/book) or in the [standard library docs](https://doc.rust-lang.org/std) anywhere else. Because "nil" is often used in programming languages to refer to "None" or "null" I think it could be a little confusing for newcomers to see this type referred to as "nil".

### CODE REVIEW

**Review by LukasKalbertodt — APPROVED** [review]
I'm in favor of this change. These docs were added 5 years ago, but that was likely just moving them from somewhere else. A quick google search also didn't bring up any recent mentions of `nil` in context of Rust referring to the unit type. I only found results from 2011 and 2012. 

What I'd like to get mentioned is `void` from languages like Java, C++ etc. I think it would be useful for people coming from those languages to know that `()` is basically equivalent to `void` (though it's a full type, unlike in some other languages). And yes I know that in languages like Haskell, `void` means something different (namely, bottom = never type). Maybe we can mention the equivalents to `()` in several languages?

We can include those changes in a new PR though. I'd just merge this when CI is green.

**Comment by rossmacarthur** [comment]
@LukasKalbertodt not sure if its possible for me to tell bors to merge? Does bors use GitHub PR approvals at all? CI is green now though.

**Comment by LukasKalbertodt** [comment]
@rossmacarthur Sorry for being imprecise. What I wanted to say is: "once the CI is green, I will merge". But then I went on a bike tour and @Mark-Simulacrum was faster. Thanks!

**Pull request merged**
